src/libs/auth.js

```diff
@@ -309,10 +309,9 @@ authStore.subscribe(withErrorReporting('Error checking registration', async (sta
       }
     }
   }
-  // need to guard against state.termsOfService not being initialized
-  const oldStatePermitsSystemUsage = oldState.termsOfService && oldState.termsOfService.permitsSystemUsage
-  const newStatePermitsSystemUsage = state.termsOfService && state.termsOfService.permitsSystemUsage
-  if ((!oldState.isSignedIn && state.isSignedIn) || (!oldStatePermitsSystemUsage && newStatePermitsSystemUsage)) {
+  const canNowUseSystem = !oldState?.termsOfService?.permitsSystemUsage && state?.termsOfService?.permitsSystemUsage
```

Comment:
Nit: `oldState` and `state` should always be defined, so no need for a `?` on them.